### PR TITLE
Refactoring Workload Stable Stack Images

### PR DESCRIPTION
**Pull request by parambole**
# Description

- Refactored workloads on jax stable stack image Names to be in line with the standard 
- Refactored stable stack variable name for more clarity

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run one-shot tests and provided workload links above if applicable. 
- [X] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

**Review by jonb377 — APPROVED** [review]
LGTM

File: dags/vm_resource.py [hunk 1]
```diff
@@ -248,17 +248,17 @@ class DockerImage(enum.Enum):
...
-      "us-docker.pkg.dev/tpu-prod-env-multipod/maxtext-jax-stable-stack/tpu:"
+  MAXTEXT_TPU_JAX_STABLE_STACK = (
+      "us-docker.pkg.dev/tpu-prod-env-multipod/maxtext_jax_stable_stack_0.4.33:"
       f"jax0.4.30-rev1-{datetime.datetime.today().strftime('%Y-%m-%d')}"
   )
-  MAXDIFFUSION_TPU_STABLE = (
-      "us-docker.pkg.dev/tpu-prod-env-multipod/maxdiffusion-jax-stable-stack/tpu:"
-      f"jax0.4.30-rev1-{datetime.datetime.today().strftime('%Y-%m-%d')}"
+  MAXDIFFUSION_TPU_JAX_STABLE_STACK = (
+      "gcr.io/tpu-prod-env-multipod/maxdiffusion_jax_stable_stack_0.4.33:"
```
Comment (RissyRan) [inline dags/vm_resource.py:256]: Please note that container registry will be deprecated in March, 2025.

**Review by RissyRan — APPROVED** [review]
LGTM! Thanks! Just have 1 note.

File: dags/vm_resource.py (hunk 1 above)
Comment (parambole) [reply dags/vm_resource.py:256]: Created https://github.com/google/maxtext/issues/904 for tracking the move.

**Pull request merged**